### ISSUE TITLE
Automatically detect compiler when not specified

### ISSUE THREAD

**Issue by SunBlack** [mozilla/sccache#1063]
The readme recommend to use `CMAKE_<LANG>_COMPILER_LAUNCHER` to use sccache. But this command [is only processed](cmake.org/cmake/help/latest/prop_tgt/LANG_COMPILER_LAUNCHER.html) in case you are using Makefile or Ninja.

For MSVC there is another way, which worked with `clcache` (which is not maintained anymore):
```bat
call "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvars64.bat"
msbuild.exe myproject.sln /p:Configuration=Release /p:CLToolExe=clcache.exe /p:CLToolPath="C:/Program Files/Python38/Scripts"
```

When using sccache we could use:
```bat
call "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Auxiliary/Build/vcvars64.bat"
msbuild.exe myproject.sln /p:Configuration=Release /p:CLToolExe=sccache.exe
```

But this lead to an invalid call:
```
C:\Users\myuser\scoop\shims\sccache.exe /c /I...
sccache : error : failed to execute compile [C:\path_to.vcxproj]
  sccache: caused by: cannot find binary path
```

Correct would be 
```
C:\Users\myuser\scoop\shims\sccache.exe cl.exe /c /I...
sccache : error : failed to execute compile [C:\path_to.vcxproj]
  sccache: caused by: cannot find binary path
```

To bypass this I tried `/p:CLToolExe="sccache.exe cl.exe"`, but this leads to:
```
error MSB6004: The specified task executable location "C:/Users/myuser/scoop/shims\sccache.exe cl.exe" is invalid.
```

So it seems there is currently no way to use this way, But I think this could be easily fixed when sccache automatically detects the compiler from the path.

**Comment by ColoredCarrot** [comment]
Just stumbled upon this (3 years later, hehe). @SunBlack did you end up finding a workaround?

**Comment by SunBlack** [comment]
Yes, by using `ccache` instead of `sccache`, because unfortunately here is no progress. Since `ccache` has been able to cache MSVC builds since version 4.6 (MSVC 2017 does not work for us, but 2019 and 2022 do if you have not activated static code analysis) and `ccache` can also be configured with a Redis backend, for example, I currently see no reason why you should use sccache if you only use C/C++.<

**Comment by ColoredCarrot** [comment]
Gotcha, thank you. That's unfortunate. `ccache` has been working excellently for us thus far (except for PCH), so we'll just stick with it and ditch `sccache`.

Thank you again for the quick answer!
